x-pack/plugins/ml/public/application/components/job_selector/use_job_selection.ts

```diff
@@ -25,49 +25,55 @@ function getInvalidJobIds(jobs: MlJobWithTimeRange[], ids: string[]) {
   });
 }
 
-function warnAboutInvalidJobIds(invalidIds: string[]) {
-  if (invalidIds.length > 0) {
-    const toastNotifications = getToastNotifications();
-    toastNotifications.addWarning(
-      i18n.translate('xpack.ml.jobSelect.requestedJobsDoesNotExistWarningMessage', {
-        defaultMessage: `Requested
-{invalidIdsLength, plural, one {job {invalidIds} does not exist} other {jobs {invalidIds} do not exist}}`,
-        values: {
-          invalidIdsLength: invalidIds.length,
-          invalidIds: invalidIds.join(),
-        },
-      })
-    );
-  }
-}
-
 export interface JobSelection {
   jobIds: string[];
   selectedGroups: string[];
 }
 
-export const useJobSelection = (jobs: MlJobWithTimeRange[], dateFormatTz: string) => {
+export const useJobSelection = (jobs: MlJobWithTimeRange[]) => {
   const [globalState, setGlobalState] = useUrlState('_g');
+  const { toasts: toastNotifications } = useNotifications();
+
+  const tmpIds = useMemo(() => {
+    const ids = globalState?.ml?.jobIds || [];
+    return (typeof ids === 'string' ? [ids] : ids).map((id: string) => String(id));
+  }, [globalState?.ml?.jobIds]);
+
+  const invalidIds = useMemo(() => {
+    return getInvalidJobIds(jobs, tmpIds);
+  }, [tmpIds]);
 
-  const jobSelection: JobSelection = { jobIds: [], selectedGroups: [] };
+  const validIds = useMemo(() => {
+    const res = difference(tmpIds, invalidIds);
+    res.sort();
+    return res;
+  }, [tmpIds, invalidIds]);
 
-  const ids = globalState?.ml?.jobIds || [];
-  const tmpIds = (typeof ids === 'string' ? [ids] : ids).map((id: string) => String(id));
-  const invalidIds = getInvalidJobIds(jobs, tmpIds);
-  const validIds = difference(tmpIds, invalidIds);
-  validIds.sort();
+  const jobSelection: JobSelection = useMemo(() => {
+    const jobIds = validIds;
+    const selectedGroups = globalState?.ml?.groups ?? [];
 
-  jobSelection.jobIds = validIds;
-  jobSelection.selectedGroups = globalState?.ml?.groups ?? [];
+    return { jobIds, selectedGroups };
```

Comment:
Nit: can't this just return `{ jobIds: validIds, selectedGroups }` - is it necessary to set `validIds` as `jobIds` on line 53?

Reply:
yes, can be written like this. I just was refactoring and kept some code untouched :)

Reply:
Changed in https://github.com/elastic/kibana/pull/71082/commits/5fa4baf276770f907eacffbaf42ce061b3f279be